Snake/BusinessLayer/maze.cs

```diff
@@ -40,20 +52,87 @@ public maze(int w,int h)
                 {
                     for (int j = 0; j < width; j++)
                     {
-                        maze[i, j] = border;
-                    } 
+                        Maze[i, j] = BORDER;
+                    }
                 }
 
                 //first and last colum of other rows should be marked as border
                 else
                 {
-                    maze[i, 0] = border;
-                    maze[i, width - 1] = border;
+                    Maze[i, 0] = BORDER;
+                    Maze[i, width - 1] = BORDER;
                 }
             }
 
-            return maze;
         }
 
+
+
+        public void Generate_All_Obstacles()
+        {
+            int Vertical_Line_Head_X = random_number.Next(0,height);
+            int Vertical_Line_Head_Y = random_number.Next(width/2,width);
+
+            Generate_Vertical_Line_Obstacle(Vertical_Line_Head_X,Vertical_Line_Head_Y,random_number.Next(0,height - Vertical_Line_Head_X));
```

Comment:
I would have a variable for the length - you have them for the X and Y, so may as weel have it for the length.